Model/Api/DiscountCodeValidation.php

```diff
@@ -444,21 +538,27 @@ private function sendErrorResponse($errCode, $message, $httpStatusCode, $quote =
         ];
         if ($quote) $errResponse['cart'] = $this->getCartTotals($quote);
         $this->response->setHttpResponseCode($httpStatusCode);
+        $this->bugsnag->notifyError('Discount Validation Response', json_encode($errResponse));
         $this->response->setBody(json_encode($errResponse));
         $this->response->sendResponse();
-        return;
+
+        return $errResponse;
     }
 
     /**
      * @param array $result
      * @param Quote $quote
+     * @return array
      * @throws \Exception
      */
-    private function sendSuccessResponse($result, $quote) {
+    private function sendSuccessResponse($result, $quote)
+    {
         $result['cart'] = $this->getCartTotals($quote);
+        $this->bugsnag->notifyError('Discount Validation Response', json_encode($result));
```

Comment:
I'm sure it's for debugging; please make sure you remove it

Reply:
I was sure that I removed it.